drivers/usb/device/usb_dc_sam_usbc_priv.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2020 Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Helper macros to make it easier to work with endpoint numbers */
+#define EP_ADDR2IDX(ep)         ((ep) & ~USB_EP_DIR_MASK)
+#define EP_ADDR2DIR(ep)         ((ep) & USB_EP_DIR_MASK)
+#define EP_UDINT_MASK()         (0x000FF000)
+
+#define NUM_OF_EP_MAX           DT_INST_PROP(0, num_bidir_endpoints)
+#define USBC_RAM_ADDR           DT_REG_ADDR(DT_NODELABEL(sram1))
+#define USBC_RAM_SIZE           DT_REG_SIZE(DT_NODELABEL(sram1))
+#define USBC_is_frozen_clk()    (USBC->USBCON & USBC_USBCON_FRZCLK)
+#define USBC_freeze_clk()       (USBC->USBCON |= USBC_USBCON_FRZCLK)
+#define USBC_unfreeze_clk()     do {					     \
```

Comment:
Why can it not be a (maybe with ALWAYS_INLINE attribute) function? (same for all this frozen_clk macros)